backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/Elasticsearch7SearchResultExtractor.java

```diff
@@ -81,13 +81,17 @@
 		Integer took = TOOK_ACCESSOR.get( responseBody ).get();
 		Boolean timedOut = TIMED_OUT_ACCESSOR.get( responseBody ).get();
 
+		Integer remainingTimeToHardTimeout = ( hardTimeoutInMilliseconds == null ) ? null :
+				Math.toIntExact( hardTimeoutInMilliseconds ) - took;
```

Comment:
That's hardly a good way to compute the remaining time. `took` is the execution time on the server side. It ignores network transfer time completely.

The "right" way to handle these things would be to keep track of when the query started executing on the client side. Then here you would be able to know exactly how much time elapsed. That's exactly what we do in the Lucene backend, and I believe that would be the only correct solution here too.

On a side note, this solution would also allow us to solve https://hibernate.atlassian.net/browse/HSEARCH-3774 (in another PR, obviously).

Reply:
Agreed. I assigned the issue to myself.